src/linter.coffee

```diff
@@ -1,38 +1,44 @@
-_ = require 'lodash'
-async = require 'async'
-asyncHandlers = require 'async-handlers'
-fsExtra = require 'fs-extra'
-DependencyLinter = require './linter/dependency_linter'
-UsedModuleFinder = require './linter/used_module_finder'
-path = require 'path'
-
-
 class Linter
 
 
-  constructor: (@dir, {allowUnused, devFilePatterns, devScripts, ignoreFilePatterns}) ->
-    @dependencyLinter = new DependencyLinter {allowUnused, devFilePatterns, devScripts}
-    @usedModuleFinder = new UsedModuleFinder {@dir, ignoreFilePatterns}
+  constructor: (@dir, {@allowUnused, @devFilePatterns, @devScripts, @ignoreFilePatterns}) ->
 
 
-  lint: (done) ->
-    async.parallel {
-      listedModules: @getListedModules
-      usedModules: @usedModuleFinder.find
-    }, asyncHandlers.transform(@dependencyLinter.lint, done)
+  lint: (done) =>
+    async = require 'async'
+    asyncHandlers = require 'async-handlers'
+    async.auto {
+      packageJson: @getPackageJson
+      usedModules: @getUsedModules
+    }, asyncHandlers.transform(@lintModules, done)
 
 
   extractListedModules: (packageJson) ->
+    _ = require 'lodash'
     {
       dependencies: _.keys(packageJson.dependencies)
       devDependencies: _.keys(packageJson.devDependencies)
     }
 
 
-  getListedModules: (done) =>
+  lintModules: ({packageJson, usedModules}) =>
+    DependencyLinter = require './linter/dependency_linter'
+    listedModules = @extractListedModules packageJson
+    dependencyLinter = new DependencyLinter {@allowUnused, @devFilePatterns, @devScripts}
+    dependencyLinter.lint {listedModules, usedModules}
+
+
+  getPackageJson: (done) =>
+    fsExtra = require 'fs-extra'
+    path = require 'path'
     filePath = path.join @dir, 'package.json'
-    callback = asyncHandlers.transform @extractListedModules, done
-    fsExtra.readJson filePath, callback
+    fsExtra.readJson filePath, done
```

Comment:
This method could be a nice single line in the old way. Now its 4 lines, 3 of them are boilerplate and ceremony.